### PR TITLE
Add desk reference for testing

### PR DESCRIPTION
**Pull request by moritztk**

### CODE REVIEW

File: desk_reference/testing.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+### What should be covered by integration tests?
+
+TODO
+
+### What should be covered by E2E tests?
+
+1.  Happy-Paths: The default scenario featuring no exceptional or error
+    conditions.
+
```
Comment (moritztk) [inline desk_reference/testing.md]: What features? All features?

Comment (moritztk) [reply desk_reference/testing.md]: What are the concrete qualifiers that should trigger a feature to be covered by E2E instead of integration?

File: desk_reference/testing.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Testing
+
+## FAQ
+
+### What should be covered by unit tests?
+
+TODO
```
Comment (moritztk) [inline desk_reference/testing.md]: Nothing, besides special edge cases?

File: desk_reference/testing.md (hunk 1 above)
Comment (moritztk) [reply desk_reference/testing.md]: Would love to here your input on this one, @davebcntk

Comment (davebcntk) [reply desk_reference/testing.md]: All the features that a user can view should at least be covered by the happy, why? because FE may or may not work depending on the browser so by having e2e we are able to do multi browser testing and detect failures.

File: desk_reference/testing.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+### What must be tested?
+
+Everything that is being used by end customers in production.
+
+### What should be covered by E2E tests?
+
+> A happy path is a default scenario featuring no exceptional or error
+> conditions. [Wikipedia](https://en.wikipedia.org/wiki/Happy_path)
+
+E2E should cover Happy-Paths and Common Paths. Do NOT test everything.
```
Comment (moritztk) [inline desk_reference/testing.md]: Would an example be helpful here?

Comment (giuband) [reply desk_reference/testing.md]: Yeah, together with a small why on the "not everything please"
